src/cplscheme/CouplingScheme.hpp

```diff
@@ -98,7 +100,44 @@ class CouplingScheme {
    *
    * Does not necessarily advance in time.
    */
-  virtual void advance() = 0;
+  void advance()
+  {
+    firstExchange();
+    secondExchange();
+  }
```

Comment:
Why is this needed?

Reply:
This provides backwards compatibility for the unit tests of cplscheme. They will be ported once we finalize the interface.